### PR TITLE
[Feature](string-function) Add function mask/mask_first_n/mask_last_n

### PR DESCRIPTION
**Pull request by mrhhsg**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Implementation of mask function from [hive](https://cwiki.apache.org/confluence/display/hive/languagemanual+udf#LanguageManualUDF-DataMaskingFunctions).

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: gensrc/script/doris_builtins_functions.py [hunk 1]
```diff
@@ -2054,6 +2054,9 @@
     [['substr', 'substring'], 'VARCHAR', ['VARCHAR', 'INT', 'INT'],
         '_ZN5doris15StringFunctions9substringEPN'
         '9doris_udf15FunctionContextERKNS1_9StringValERKNS1_6IntValES9_', '', '', 'vec', 'ALWAYS_NULLABLE'],
+    [['mask'], 'STRING', ['STRING', '...'], '', '', '', 'vec', 'CUSTOM'],
```
Comment (HappenLee) [inline gensrc/script/doris_builtins_functions.py:2057]: seems the function is `default` not `custom`?

File: be/src/vec/functions/function_string.h [hunk 2]
```diff
@@ -302,6 +302,211 @@ struct Substr2Impl {
...
+
+        ColumnPtr source_column;
+        auto res = ColumnString::create();
+        source_column = block.get_by_position(arguments[0]).column;
+        if (source_column->is_nullable()) {
+            auto* nullable = assert_cast<const ColumnNullable*>(source_column.get());
+            VectorizedUtils::update_null_map(null_map->get_data(), nullable->get_null_map_data());
+            source_column = nullable->get_nested_column_ptr();
+        }
+
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.h:344]: maybe return error the other column is not `const`

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.08 seconds
 load time: 601 seconds
 storage size: 17154875712 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221027003932_clickbench_pr_34486.html

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.83 seconds
 load time: 563 seconds
 storage size: 17154644722 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028010814_clickbench_pr_34993.html

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**